client/components/candidate-card/index.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, ConfigOptions, AttrsSchema, h } from 'panel';
+
+import { safelyDefine } from '../../utils';
+
+import { CandidateCardAttrs, CandidateCardState } from './types';
+import css from './style.scss';
+
+const componentName = 'candidate-card';
+export default class DelegateCard extends Component<
+  CandidateCardState,
+  {},
+  unknown,
+  CandidateCardAttrs
+> {
+  static get attrsSchema(): AttrsSchema<CandidateCardAttrs> {
+    return {
+      ...super.attrsSchema,
+      name: 'string',
+    };
+  }
+
+  get config(): ConfigOptions<CandidateCardState, {}, CandidateCardAttrs> {
+    return {
+      css,
+      useShadowDom: true,
+      defaultState: {
+        showDetails: false,
+      },
+      helpers: {},
+      template: ({ $attr, $helpers, showDetails }) => {
+        if (showDetails) {
+          return h('div', 'Details');
+        }
+        return h('slot');
```

Comment:
If reading through the SSR note, you ask why are we slotting at all, here is the beginning of the plan. Slot the initial/default display, but then replace that when the user interacts with the card. So basically the "details" wouldn't be SSR'd but the normal view is.